src/snapshot/patch_apply.rs

```diff
@@ -117,24 +117,24 @@ fn apply_add_child(
     context: &mut PatchApplyContext,
     tree: &mut RojoTree,
     parent_id: Ref,
-    snapshot: InstanceSnapshot,
+    mut snapshot: InstanceSnapshot,
 ) {
     let snapshot_id = snapshot.snapshot_id;
-    let properties = snapshot.properties;
-    let children = snapshot.children;
-
-    // Property application is deferred until after all children
-    // are constructed. This helps apply referents correctly.
-    let remaining_snapshot = InstanceSnapshot::new()
-        .name(snapshot.name)
-        .class_name(snapshot.class_name)
-        .metadata(snapshot.metadata)
-        .snapshot_id(snapshot.snapshot_id);
-
-    let id = tree.insert_instance(parent_id, remaining_snapshot);
+    let children = take(&mut snapshot.children);
+
+    // If an object we're adding has a non-null referent, we'll note this
+    // instance down as needing to be revisited later.
+    let has_refs = snapshot.properties.values().any(|value| match value {
+        Variant::Ref(value) => value.is_some(),
```

Comment:
`matches!(value, Variant::Ref(Some(_))` Clippy lints this, but it might not check is_some() checks

Reply:
I think this is a clippy false positive. `value` is not an `Option`, it's actually just a `Ref`, which has `is_some` and `is_none` methods.

Reply:
Oh, I meant that clippy *should* catch this, under my assumption it was an Option. My mistake.

Reply:
That's probably an indication that these method names are bad, but I didn't know what else to call them, lol!